avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_4/Array_of_UNION_GenericDeserializer_585074122056792963_585074122056792963.java

```diff
@@ -67,16 +70,22 @@ public IndexedRecord deserializerecord0(Object reuse, Decoder decoder)
             record = new org.apache.avro.generic.GenericData.Record(arrayElemOptionSchema0);
         }
         int unionIndex1 = (decoder.readIndex());
-        if (unionIndex1 == 0) {
-            decoder.readNull();
-        } else {
-            if (unionIndex1 == 1) {
-                if (record.get(0) instanceof Utf8) {
-                    record.put(0, (decoder).readString(((Utf8) record.get(0))));
+        switch (unionIndex1) {
+            case  0 :
+                decoder.readNull();
+                break;
+            case  1 :
+            {
+                Object oldString0 = record.get(0);
+                if (oldString0 instanceof Utf8) {
+                    record.put(0, (decoder).readString(((Utf8) oldString0)));
```

Comment:
@radai-rosenblatt this is another thing you asked for so that the get invocation happens only once, rather than twice...